File: Backend/app/schemas/oferta.py
from datetime import date
from decimal import Decimal

from pydantic import BaseModel, ConfigDict, Field


class OfertaExistenteCreate(BaseModel):
    """Crea una oferta sobre un producto que ya existe."""
    producto_id: int
    nombre: str
    descuento: Decimal = Field(ge=0, le=100)
    fecha_inicio: date
    fecha_fin: date
    estado: str = "Activa"


class OfertaUpdate(BaseModel):
    nombre: str | None = None
    descuento: Decimal | None = Field(default=None, ge=0, le=100)
    fecha_inicio: date | None = None
    fecha_fin: date | None = None
    estado: str | None = None


class OfertaResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    nombre: str
    descuento: Decimal
    fecha_inicio: date
    fecha_fin: date
    estado: str
    producto_id: int
    imagen_url: str | None = None
    nombre_producto: str | None = None
    precio_original: Decimal | None = None
    stock_producto: int | None = None
    fecha_vencimiento: date | None = None

File: App-android/app/src/main/java/com/example/app/modelos/Oferta.kt
package com.example.app.modelos

import com.google.gson.annotations.SerializedName

data class Oferta(
    val id: Int,
    val nombre: String,
    val descuento: Double,
    @SerializedName("fecha_inicio")
    val fechaInicio: String,
    @SerializedName("fecha_fin")
    val fechaFin: String,
    val estado: String,
    @SerializedName("producto_id")
    val productoId: Int,
    @SerializedName("imagen_url")
    val imagenUrl: String?,
    @SerializedName("nombre_producto")
    val nombreProducto: String?,
    @SerializedName("precio_original")
    val precioOriginal: Double?,
    @SerializedName("stock_producto")
    val stockProducto: Int?,
    // Este campo lo obtendremos del producto asociado si es necesario, 
    // pero el backend OfertaResponse ya trae los datos básicos.
    // Para la fecha de vencimiento, necesitaremos que el backend la incluya o hacer un join extra.
    @SerializedName("fecha_vencimiento")
    val fechaVencimiento: String? = null
)
